src/community/geofence-server/src/main/java/org/geoserver/geofence/internal/InternalUserResolver.java

```diff
@@ -29,12 +34,54 @@
  * @author Niels Charlier
  *
  */
+@PropertySource("classpath*:application.properties")
 public class InternalUserResolver implements UserResolver {
     
-    private Logger logger = Logging.getLogger(InternalUserResolver.class);
+    public static final String DEFAULT_USER_GROUP_SERVICE_KEY = "org.geoserver.rest.DefaultUserGroupServiceName";
+
+	private Logger logger = Logging.getLogger(InternalUserResolver.class);
+    
+    @Autowired
+    private Environment env;
+    
+    @Value("${org.geoserver.rest.DefaultUserGroupServiceName}")
+    private String DEFAULT_ROLE_SERVICE_NAME;
+    
+    private GeoServerSecurityService defaultSecurityService;
     
     protected GeoServerSecurityManager securityManager;
     
+    private String getDefaultServiceName() {
+        String defaultServiceName = System.getProperty(DEFAULT_USER_GROUP_SERVICE_KEY);
+        defaultServiceName = defaultServiceName == null ? env.getProperty(DEFAULT_USER_GROUP_SERVICE_KEY) : DEFAULT_ROLE_SERVICE_NAME;
+        return defaultServiceName == null ? DEFAULT_ROLE_SERVICE_NAME : defaultServiceName;
+    }
+    
+    public GeoServerSecurityService getDefaultSecurityService() throws IOException {
```

Comment:
All this new logic assigning in a mutex way groupservice or roleservice would require some testing...

Reply:
Testing as in manual testing or testing as in unit testing?